Restructuring the documentation: cleanup in operational (#5465)
OpenNMS/opennms: docs/modules/reference/pages/configuration/ifttt-integration.adoc

= IFTTT Integration

The IFTTT service lets you combine web applications using simple conditional instructions.
Each supported service has several triggers that you can use to trigger the actions of other services.

> This lets you, for example, change brightness and color of a smart bulb, and send messages or date to IoT devices.

Reviewer: ```suggestion
This lets you, for example, change the brightness and color of a smart bulb, and send messages or the date to IoT devices.
```

The {page-component-title} integration uses the Webhooks service, that lets you trigger actions when a specific web request is received.
The basic operation is as follows: {page-component-title} polls for alarms and matches the alarm reduction key against a given filter and the alarm's associated nodes against a given category filter.

> For the resulting alarm, set the maximum severity and the total count is computed.

Reviewer: ```suggestion
For the resulting alarm, set the maximum severity to compute the total count.
```

If one of these values has changed compared to the last poll, one or more events specified for the computed maximum severity will be sent to IFTTT.

== IFTTT configuration

To use the IFTTT integration in {page-component-title} you need an IFTTT account.
With this account you can create applets that combine a trigger with an action.
In our case, we use the Webhooks service as the trigger and define the event name `OpenNMS`.

> After this step you can combine this trigger with any of the possible supported services and their actions.

Reviewer: ```suggestion
After doing so, you can combine this trigger with any of the possible supported services and their actions.
```
Or does the original sentence mean the procedure described below?

[[ga-ifttt-trigger-definition]]
.Webhooks service trigger definition
image::alarms/trigger-definition-small.png[]

> In your account service settings for the Webhooks service you find your key in the given service URL.

Reviewer: ```suggestion
In your account service settings for the Webhooks service, find your key in the given service URL.
```

In the following example this key is `X71dfUZsH4Wkl6cjsLjdV`.

[[ga-ifttt-webhooks-settings]]
.Webhooks service settings
image::alarms/webhooks-settings-small.png[]

> On the side of {page-component-title}, you need a configuration that defines which event names to send on an alarm count or severity change.

Reviewer: ```suggestion
In {page-component-title}, specify a configuration that defines which event names to send on an alarm count or severity change.
```

> The configuration file `ifttt-config.xml` contains trigger packages.

Reviewer: ```suggestion
The configuration file `ifttt-config.xml` contains trigger packages that work as follows:
```

> The operation is as follows:

Reviewer: ```suggestion

```

> . {page-component-title} retrieves all alarms that have a node associated.

Reviewer: This section is not showing up as steps in the final.
It's also not clear if this is steps the user can do to configure this behaviour of if this text is just describing something that happens.

. Each trigger package defines whether to take into account only acknowledged alarms.

> . Compute the maximum severity and alarm count for each trigger package's category filter and reduction key filter.

Reviewer: ```suggestion
. {page-component-title} then computes the maximum severity and alarm count for each trigger package's category filter and reduction key filter.
```

> . Trigger all events defined in the corresponding trigger sets for the computed maximum severity.

Reviewer: ```suggestion
. It triggers all events defined in the corresponding trigger sets for the computed maximum severity.
```

The category and reduction key filter accepts Java regular expressions.

> Using an empty category filter will use all unacknowledged alarms regardless of whether these alarms have nodes assigned.

Reviewer: ```suggestion
Using an empty category filter will use all unacknowledged alarms regardless of whether they have nodes assigned.
```

Each trigger inside a trigger set defines the event name to be triggered and three additional values.
Use these values to set additional attributes for the corresponding IFTTT applet action.
You can define the following trigger sets:

[options="header"]
[cols="1,3"]
|===
| Name     | Execution

> | ON       | on start of the IFTTT alarm polling daemon to switch on a device

Reviewer: ```suggestion
| ON       | When the IFTTT alarm starts, the polling daemon switches on a device.
```
I'm not sure if my suggestion is correct, but the original did not make sense.

> | OFF      | on stop of the IFTTT alarm polling daemon to switch off a device

Reviewer: ```suggestion
| OFF      | When the IFTTT alarm stops, the polling daemon switches off a device.
```

| NORMAL   | if severity is NORMAL
| WARNING  | if severity is WARNING
| MINOR    | if severity is MINOR
| MAJOR    | if severity is MAJOR
| CRITICAL | if severity is CRITICAL
|===

You can use the ON and OFF events when initializing devices or powering them up/down.

== OpenNMS configuration

> IFTTT alarm polling will be enabled by setting the attribute `enabled` to `true` in the `ifttt-config.xml` file.

Reviewer 2: ```suggestion
IFTTT alarm polling will be enabled by setting the attribute `enabled` to `true` in `ifttt-config.xml`.
```

It is also possible to configure the polling interval.

> The following trigger package defines the trigger sets which themselves define a sequence of events to be triggered at IFTTT.

Reviewer 2: ```suggestion
The following package defines the trigger sets which set a sequence of events to be triggered at IFTTT.
```

Each trigger defines the eventName and an additional delay.

> This lets you defer the execution of the next trigger in a trigger set.

Reviewer 2: ```suggestion
This lets you defer the next trigger in a trigger set.
```

== Example

> The following example shows the configuration file for a WiFi light bulb controlled via IFTTT.

Reviewer 2: ```suggestion
The following example shows the configuration file for a Wi-Fi light bulb controlled via IFTTT.
```

> The defined applets use `value1` for setting the color and `value2` for setting the brightness.

Reviewer 2: ```suggestion
The defined applets use `value1` to set the color and `value2` to set the brightness.
```

The third value demonstrates the use of placeholders.
For the severity-based trigger sets, you can use the following placeholders in the three value fields:
`%os%`/`%oldSeverity` for old severity, `%ns%`/`%newSeverity%` for new severity, `%oc%`/`%oldCount` for old alarm count and `%nc%`/`%newCount% for new alarm count.
This is useful for sending messages or operating LED displays via IFTTT.

> [source, xml]

Reviewer 2: ```suggestion
.Example IFTTT configuration
[source, xml]
```
